llarp/util/thread/threading.hpp

```diff
@@ -31,9 +31,16 @@ namespace llarp
   namespace util
   {
     /// a mutex that does nothing
+    /// 
+    /// this exists to convert "old" mutexes that may no longer be necessary
```

Comment:
nit: I would note that previously they were *real* mutexes too.

Reply:
added even more comments in a3c48e2